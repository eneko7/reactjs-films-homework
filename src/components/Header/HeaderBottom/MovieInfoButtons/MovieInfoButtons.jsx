import React from 'react';
import style from './MovieInfoButtons.scss';

class MovieInfoButtons extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            isOpen: false,
        };
        this.openInfoClick = this.openInfoClick.bind(this);
    }
    openInfoClick() {
        this.setState(prevState => ({
            isOpen: !prevState.isOpen
        }));
    }
    render () {
        let active = this.state.isOpen?`${style.active}`:'';
        return (
            <div className={style.bottom_right}>
                <div className={`${active} ${style.bottom_right_text_descr}`} >
                    <p className={style.bottom_right_text_descr_p}>
                        {this.props.annotation}
                    </p>
                </div>
                <div className={`${active} ${style.bottom_right_buttons_box}`}>
                    <button className={`${style.bottom_right_buttons_box_button_watch} ${style.bottom_right_buttons_box_button}`}>Watch Now</button>
                    <button className={`${style.bottom_right_buttons_box_button_view} ${style.bottom_right_buttons_box_button}`} onClick={this.openInfoClick}>{!this.state.isOpen?'View Info':'Close'}</button>
                </div>
            </div>
        );
    }
}

export default MovieInfoButtons;