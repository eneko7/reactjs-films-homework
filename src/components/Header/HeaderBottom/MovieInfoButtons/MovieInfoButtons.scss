.bottom_right {
  width: 370px;
  margin-top: 122px;

  .bottom_right_text_descr {
    width: 350px;
    font-size: 20px;
    line-height: 32px;
    color: #fff;
    font-family: Helvetica, Arial, Verdana, Tahoma, sans-serif;
    position: relative;
    top: -50px;
    left: 30px;
    background: rgba(0, 0, 0, 0.5);
    padding: 2px 0 2px 25px;
    margin-left: 25px;
    display: none;
  }

  .bottom_right_text_descr.active {
    display: block;
  }

  .bottom_right_buttons_box {
    display: flex;
    justify-content: space-between;
    margin-right: 30px;
    position: relative;
    top: 200px;

    .bottom_right_buttons_box_button {
      font-size: 18px;
      letter-spacing: 1px;
      line-height: 30px;
      color: #fff;
      font-family: Helvetica, Arial, Verdana, Tahoma, sans-serif;
      text-align: center;
      width: 150px;
      height: 40px;
      border-radius: 6px;
      cursor: pointer;
    }

    .bottom_right_buttons_box_button:focus {
      outline: none;
    }

    .bottom_right_buttons_box_button_watch {
      background-color: #3bb3df;
      border-style: none;
    }

    .bottom_right_buttons_box_button_view {
      background: none;
    }

    .bottom_right_buttons_box_button_view:hover {
      background: #3bb3df;
      border-style: none;
    }
  }

  .bottom_right_buttons_box.active {
    position: relative;
    border-style: none;
    top: -4px;
  }
}

@media (max-width: 767px) {
  .bottom_right {
    width: 100%;
    margin: 0;

    .bottom_right_text_descr {
      width: 100%;
      font-size: 16px;
      line-height: 20px;
      color: #fff;
      font-family: Helvetica, Arial, Verdana, Tahoma, sans-serif;
      top: 0;
      left: -9%;
      background: rgba(0, 0, 0, 0.5);
      padding: 2px 0 2px 25px;
      margin-left: 0;
    }

    .bottom_right_buttons_box {
      margin-right: 0;
      position: relative;
      top: 0;
      width: 100%;

      .bottom_right_buttons_box_button {
        width: 130px;
        font-size: 17px;
      }

      .bottom_right_buttons_box_button_view {
        margin-right: 30px;
      }
    }

    .bottom_right_buttons_box.active {
      top: 0;
    }
  }
}
