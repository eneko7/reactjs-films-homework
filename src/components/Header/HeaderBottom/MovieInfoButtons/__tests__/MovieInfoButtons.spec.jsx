import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import renderer from 'react-test-renderer';
import MovieInfoButtons from '../index';

jest.mock('../../../../MoviesGrid/MovieElement/ModalWindowFilm/ModalWindowFilmContainer', () => () => <div>test</div>);

const data = {
  annotation: 'film description',
  filmId: 1023,
};

describe('MovieInfoButtons', () => {
  it('renders correctly', () => {
    const shallowRenderer = new ShallowRenderer();
    shallowRenderer.render(
      <MovieInfoButtons {...data} />,
    );
    const result = shallowRenderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });

  it('MovieInfoButtons -> Click (show info) ', () => {
    const output = renderer.create(<MovieInfoButtons {...data} />);
    output.root.findByProps({ className: 'bottom_right_buttons_box_button_view bottom_right_buttons_box_button' }).props.onClick();
    expect(output).toMatchSnapshot();
  });

  it('MovieInfoButtons -> Click (close info) ', () => {
    const output = renderer.create(<MovieInfoButtons {...data} />);
    output.root.findByProps({ className: 'bottom_right_buttons_box_button_view bottom_right_buttons_box_button' }).props.onClick();
    output.root.findByProps({ className: 'bottom_right_buttons_box_button_view bottom_right_buttons_box_button' }).props.onClick();
    expect(output).toMatchSnapshot();
  });

  it('MovieInfoButtons -> Click (watch film) ', () => {
    const output = renderer.create(<MovieInfoButtons {...data} />);
    output.root.findByProps({ className: 'bottom_right_buttons_box_button_watch bottom_right_buttons_box_button' }).props.onClick();
    expect(output).toMatchSnapshot();
  });

  it('MovieInfoButtons -> Click (close film) ', () => {
    const output = renderer.create(<MovieInfoButtons {...data} />);
    output.root.findByProps({ className: 'bottom_right_buttons_box_button_watch bottom_right_buttons_box_button' }).props.onClick();
    output.root.findByProps({ className: 'bottom_right_buttons_box_button_watch bottom_right_buttons_box_button' }).props.onClick();
    expect(output).toMatchSnapshot();
  });
});
