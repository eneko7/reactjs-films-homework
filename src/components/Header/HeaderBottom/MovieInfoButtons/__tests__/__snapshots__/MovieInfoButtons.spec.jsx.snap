// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieInfoButtons MovieInfoButtons -> Click (close film)  1`] = `
<div
  className="bottom_right"
>
  <div
    className=" bottom_right_text_descr"
  >
    <p
      className="bottom_right_text_descr_p"
    >
      film description
    </p>
  </div>
  <div
    className=" bottom_right_buttons_box"
  >
    <button
      className="bottom_right_buttons_box_button_watch bottom_right_buttons_box_button"
      onClick={[Function]}
      type="button"
    >
      Watch Now
    </button>
    <button
      className="bottom_right_buttons_box_button_view bottom_right_buttons_box_button"
      onClick={[Function]}
      type="button"
    >
      View Info
    </button>
  </div>
</div>
`;

exports[`MovieInfoButtons MovieInfoButtons -> Click (close info)  1`] = `
<div
  className="bottom_right"
>
  <div
    className=" bottom_right_text_descr"
  >
    <p
      className="bottom_right_text_descr_p"
    >
      film description
    </p>
  </div>
  <div
    className=" bottom_right_buttons_box"
  >
    <button
      className="bottom_right_buttons_box_button_watch bottom_right_buttons_box_button"
      onClick={[Function]}
      type="button"
    >
      Watch Now
    </button>
    <button
      className="bottom_right_buttons_box_button_view bottom_right_buttons_box_button"
      onClick={[Function]}
      type="button"
    >
      View Info
    </button>
  </div>
</div>
`;

exports[`MovieInfoButtons MovieInfoButtons -> Click (show info)  1`] = `
<div
  className="bottom_right"
>
  <div
    className="active bottom_right_text_descr"
  >
    <p
      className="bottom_right_text_descr_p"
    >
      film description
    </p>
  </div>
  <div
    className="active bottom_right_buttons_box"
  >
    <button
      className="bottom_right_buttons_box_button_watch bottom_right_buttons_box_button"
      onClick={[Function]}
      type="button"
    >
      Watch Now
    </button>
    <button
      className="bottom_right_buttons_box_button_view bottom_right_buttons_box_button"
      onClick={[Function]}
      type="button"
    >
      Close
    </button>
  </div>
</div>
`;

exports[`MovieInfoButtons MovieInfoButtons -> Click (watch film)  1`] = `
Array [
  <div
    className="bottom_right"
  >
    <div
      className=" bottom_right_text_descr"
    >
      <p
        className="bottom_right_text_descr_p"
      >
        film description
      </p>
    </div>
    <div
      className=" bottom_right_buttons_box"
    >
      <button
        className="bottom_right_buttons_box_button_watch bottom_right_buttons_box_button"
        onClick={[Function]}
        type="button"
      >
        Watch Now
      </button>
      <button
        className="bottom_right_buttons_box_button_view bottom_right_buttons_box_button"
        onClick={[Function]}
        type="button"
      >
        View Info
      </button>
    </div>
  </div>,
  <div>
    test
  </div>,
]
`;

exports[`MovieInfoButtons renders correctly 1`] = `
<React.Fragment>
  <div
    className="bottom_right"
  >
    <div
      className=" bottom_right_text_descr"
    >
      <p
        className="bottom_right_text_descr_p"
      >
        film description
      </p>
    </div>
    <div
      className=" bottom_right_buttons_box"
    >
      <button
        className="bottom_right_buttons_box_button_watch bottom_right_buttons_box_button"
        onClick={[Function]}
        type="button"
      >
        Watch Now
      </button>
      <button
        className="bottom_right_buttons_box_button_view bottom_right_buttons_box_button"
        onClick={[Function]}
        type="button"
      >
        View Info
      </button>
    </div>
  </div>
</React.Fragment>
`;
